checkov/common/util/prompt.py

```diff
@@ -0,0 +1,273 @@
+                                                "default": "resource",
+                                                "sub_prompts": [
+                                                    {
+                                                        "supported_resource": {
+                                                            "prompt_if": 'resource',
+                                                            "text": 'Enter the terraform object type',
+                                                            "typ": str,
+                                                            "default": "google_project_iam_policy"
+                                                        },
+                                                    }
+                                                ]
+                                            }
+                                        },
+                                    ]
+                                }
+                            }
+                        ]
+                    }
+                }
+            ]
+        }
+    }
+
+    def __init__(self):
+        print(checkov_banner)
+        self.responses = {}
+        self.prompt()
+
+    def go(self):
+        print("Please ensure you are at the root of the Checkov repository before completing this prompt")
+
+        # Load our prompts
+        module = importlib.import_module("checkov.common.util.prompt")
+
+        # Load an instance of our Check class (this could be expanded to do more later)
+        instance = getattr(module, "Check")(self.responses)
+
+        # Call the action on that Class (add, remove, modify - leaves room for more actions)
+        getattr(instance, self.action)()
+
+    # Recurse over our prompt, populating new class attributes from the keys and
+    # user-supplied answers
+    def prompt(self, prompt_map=PROMPTS, prompt_if=None):
+        for k, v in prompt_map.items():
+            if "prompt_if" not in v or v["prompt_if"] == prompt_if:
+                # Prompt the user
+                p = click.prompt(v["text"], type=v["typ"], default=v["default"])
+
+                # Create the action on our object
+                if k is 'action':
+                    setattr(self, k, p)
+
+                # Record user responses
+                self.responses[k] = p
+                print()  # Newline for readability
+
+                # Call prompt() again on any sub_prompts
+                if "sub_prompts" in v:
+                    for sub_prompt in v["sub_prompts"]:
+                        self.prompt(sub_prompt, prompt_if=p)
+
+    def template_env(self):
+        template_loader = jinja2.FileSystemLoader(searchpath=TEMPLATE_DIRECTORY)
+        return jinja2.Environment(loader=template_loader)
+
+
+class Check(Prompt):
+    def __init__(self, user_responses={}):
+        for k, v in user_responses.items():
+            setattr(self, k, v)
```

Comment:
it is better to do an explicit attribute assignment here, otherwise you can't be sure later that a specific attribute even exist.

Reply:
Pretty sure I addressed this how you wanted, let me know if not.